main.tf

```diff
@@ -1,236 +1,428 @@
+
+  api_id = aws_apigatewayv2_api.this[0].id
+
+  connection_id             = try(aws_apigatewayv2_vpc_link.this[each.value.vpc_link_key].id, each.value.connection_id)
+  connection_type           = each.value.connection_type
+  content_handling_strategy = each.value.content_handling_strategy
+  credentials_arn           = each.value.credentials_arn
+  description               = each.value.description
+  integration_method        = each.value.method
+  integration_subtype       = each.value.subtype
+  integration_type          = each.value.type
+  integration_uri           = each.value.uri
+  passthrough_behavior      = each.value.passthrough_behavior
+  payload_format_version    = each.value.payload_format_version
+  request_parameters        = each.value.request_parameters
+  request_templates         = each.value.request_templates
+
+  dynamic "response_parameters" {
+    for_each = coalesce(each.value.response_parameters, [])
+
+    content {
+      mappings    = response_parameters.value.mappings
+      status_code = response_parameters.value.status_code
     }
   }
 
-  dynamic "route_settings" {
-    for_each = { for k, v in var.integrations : k => v if var.create_routes_and_integrations && try(tobool(v.create_route), true) && length(setintersection(["data_trace_enabled", "detailed_metrics_enabled", "logging_level", "throttling_burst_limit", "throttling_rate_limit"], keys(v))) > 0 }
+  template_selection_expression = each.value.template_selection_expression
+  timeout_milliseconds          = each.value.timeout_milliseconds
 
-    content {
-      route_key          = route_settings.key
-      data_trace_enabled = try(route_settings.value.data_trace_enabled, var.default_route_settings["data_trace_enabled"], false) # supported in Websocket APIGateway only
-      logging_level      = try(route_settings.value.logging_level, var.default_route_settings["logging_level"], null)            # supported in Websocket APIGateway only
+  dynamic "tls_config" {
+    # For some reason the default is not using an empty obect and causing an interation to occur
+    for_each = length({ for k, v in each.value.tls_config : k => v if v != null }) > 0 ? [each.value.tls_config] : []
 
-      detailed_metrics_enabled = try(route_settings.value.detailed_metrics_enabled, var.default_route_settings["detailed_metrics_enabled"], false)
-      throttling_burst_limit   = try(route_settings.value.throttling_burst_limit, var.default_route_settings["throttling_burst_limit"], null)
-      throttling_rate_limit    = try(route_settings.value.throttling_rate_limit, var.default_route_settings["throttling_rate_limit"], null)
+    content {
+      server_name_to_verify = replace(tls_config.value.server_name_to_verify, "*.", "")
     }
   }
 
-  tags = merge(var.default_stage_tags, var.tags)
-
-  # Bug in terraform-aws-provider with perpetual diff
   lifecycle {
-    ignore_changes = [deployment_id]
```

Comment:
Is this really resolved for HTTP Api Gateway?

Reply:
I didn't see this one come up in the diffs for changes that were "drifted" - I did see two others, but not sure what the resolution is for those yet. But for this one, it does appear to be resolved (maybe because we're using the `deployment_id` now in this PR?). Here are the two that do show up from time to time (from the example): 

![image](https://github.com/terraform-aws-modules/terraform-aws-apigateway-v2/assets/10913471/0983c942-2afa-4308-8a2e-3d7e8fc39951)